File: packages/domain-git/src/git-file-status.ts
/**
 * Git file status enumeration
 */
export enum GitFileStatus {
  /** File has been added to the staging area */
  Added = 'Added',
  
  /** File has been modified */
  Modified = 'Modified',
  
  /** File has been deleted */
  Deleted = 'Deleted',
  
  /** File has been renamed */
  Renamed = 'Renamed',
  
  /** File is untracked (not in git) */
  Untracked = 'Untracked',
  
  /** File has been copied */
  Copied = 'Copied'
}

File: apps/backend/Models/GitFileStatus.cs
namespace GitRelax.Backend.Models;

/// <summary>
/// Status of a file in git
/// </summary>
public enum GitFileStatus
{
    /// <summary>
    /// File has been added to staging area
    /// </summary>
    Added,
    
    /// <summary>
    /// File has been modified
    /// </summary>
    Modified,
    
    /// <summary>
    /// File has been deleted
    /// </summary>
    Deleted,
    
    /// <summary>
    /// File has been renamed
    /// </summary>
    Renamed,
    
    /// <summary>
    /// File is untracked (not in git)
    /// </summary>
    Untracked,
    
    /// <summary>
    /// File has been copied
    /// </summary>
    Copied
}
